lib/farmsmart_localizations.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:intl/intl.dart';
+
+import 'l10n/messages_all.dart';
+
+class FarmsmartLocalizations {
```

Comment:
Not sure about this global scope class, this would mean all widgets depend on it. All widgets are now FarmSmart widgets and will need this class that has all the strings in the whole app.
It would be better to have a generic function like the Intl.message("XXXX") and inject in the translation provider.
The best place to do this is in the view models. That way the widgets only know about their view model interface and no global scope classes. For now if we could we add this system, but not implement it in any widget apart from the App bar (as a usage example). Lets have a chat around this @maribel4amido

Reply:
That's a fair point if we go with the route of committing fully to using the intl library. 
My understanding with this flutter recommended approach of having a single class and this method, is that it's possible to replace intl implementation (say if we want to use a hashmap instead). So it's very flexible. Also, I don't think the strings will be that many app wide.

The advantages that I see with this single entry point approach to localisation:
1. Flexibility of localisation library. If we change the localisation approach we only change these classes.
2. Single responsibility for defining translations and strings. At the moment the dart tool requires you to supply all classes where Intl.message is defined in order for it to be in the generated arb file. We'll have to have a smart tool to find all files with Intl.message occurrence (not impossible though)

Reply:
No this is not what I mean, we should not use the intl function directly. Instead we will need each viewModel to have a strings provider injected. this way we can change the implementation to anything we want, and not make all the widgets depend on a singleton. We can have this global class if we want, but no widget can use it directly. i.e no widget should have FarmsmartLocalizations imported.

Reply:
I get you. An abstraction layer. Since this piece is not touching any view model for the moment we'll keep this in mind when multi-language support is introduced.